src/lib/lockfile.ts

```diff
@@ -0,0 +1,34 @@
+import * as fs from 'fs';
+import { dirname } from 'path';
+
+import { exec, unlinkAll } from './fs-utils';
+
+// Equivalent to `drwxrwxrwt`
+const STICKY_WRITE_PERMISSIONS = parseInt('1777', 8);
+export const NOBODY_UID = 65534;
+
+export async function lock(path: string) {
+	/**
+	 * Set parent directory permissions to `drwxrwxrwt` (octal 1777), which are needed
+	 * for lockfile binary to run successfully as the `nobody` (UID 65534) user.
+	 * Otherwise the `nobody` user will not have write permissions to the necessary
+	 * bind-mounted lockfile directories. `chmod` does not fail or throw if the
+	 * directory already has the proper permissions.
+	 */
+	await fs.promises.chmod(dirname(path), STICKY_WRITE_PERMISSIONS);
```

Comment:
As /tmp contents are stored in memory, calling chmod on each lockfile.lock call will not affect storage media and is of negligible overhead.